packages/commons/src/auth/jwt.ts

```diff
@@ -52,10 +63,13 @@ export const verifyJwtToken = (jwtToken: string): Promise<boolean> => {
   return new Promise((resolve, _reject) => {
     jwt.verify(jwtToken, getKey(clients), undefined, function (err, _decoded) {
       if (err) {
-        logger.warn(`Error verifying token: ${err}`);
-        return resolve(false);
+        logger.warn(`Token verification fails: ${err}`);
+        return resolve({
+          valid: false,
+          error: err.message,
+        });
```

Comment:
I see that `readAuthDataFromJwtToken` throws in case of error, while this `verifyJwtToken` returns the error instead of throwing directly. Is there a reason for this difference? 
Could it make sense to follow the same pattern and either have both functions throw or both functions return a result and let the caller throw (the `authenticationMiddleware` in this case)

Reply:
This difference is deliberately implemented like this, during the first part of the refactoring I had some doubts whether to follow the same error generation pattern or use the validation result object.
Ultimately, from my point of view, the difference is how the methods are used by those who invoke the methods, reading the authentication data from JWT is an operation that in accordance with the projections returns the authentication data as expected, if an error occurs it should be handled with try-catch block and stops the flow execution;
Meanwhile for JWT verification an error is expected, so the validation result could be handled differently, using if statement for execution flow checks, instead of try-catch block.
This is a stylistic opinion, however currently in our code base verification is only used to generate errors 😅, and it is possible to generate errors again in this case... what do you think of this different approach?

Reply:
Both ways work for me! If we use this function's return value only to throw in case it's invalid, I would follow the same error pattern here and throw in case of error. I agree that for general use, this function could be more suitable, but we now use it only once and throw when validation fails